internal/dot/graph_test.go

```diff
@@ -181,6 +198,41 @@ func TestFailNodes(t *testing.T) {
 		assert.Equal(t, transitiveFailure, c1.ErrorType)
 	})
 
+	t.Run("no constructors should be removed because they all have failing results", func(t *testing.T) {
+		dg := NewGraph()
+		c0 := &Ctor{ID: 123}
+		c1 := &Ctor{ID: 456}
+
+		dg.AddCtor(c0, []*Param{}, []*Result{r1})
+		dg.AddCtor(c1, []*Param{}, []*Result{r2})
+
+		dg.FailNodes([]*Result{r1}, 123)
+		dg.FailNodes([]*Result{r2}, 456)
+		dg.PruneSuccess()
+
+		assert.Equal(t, 2, len(dg.Ctors))
+		assert.Equal(t, 2, len(dg.ctorMap))
```

Comment:
nit: can use assert.Len instead

```suggestion
		assert.Len(t, dg.ctorMap, 2)
```